File: YKOCBaseClass/Classes/YKCollectionViewCell.m
//
//  YKCollectionViewCell.m
//  YKOCBaseClass
//
//  Created by edward on 2021/8/25.
//

#import "YKCollectionViewCell.h"

@implementation YKCollectionViewCell

- (void)awakeFromNib
{
    [super awakeFromNib];
    
}

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        [self autoExecute];
    }
    return self;
}
/// 在viewdidload自动执行
- (void)autoExecute
{
    
}

/// UI构建
- (void)didSetupUI:(UIView *)view
{
    
}

- (void)didBindData
{
    
}

- (void)configDataWithViewModel:(YKViewModel *)viewModel indexPath:(NSIndexPath *)indexPath dataSource:(NSArray *)dataSource
{
    
}


@end

File: Module/YKSwiftBaseClass/YKSwiftBaseClass/Classes/YKCollectionViewCell.swift
//
//  YKCollectionViewCell.swift
//  YKSwiftBaseClass
//
//  Created by linghit on 2021/11/17.
//

import UIKit

open class YKCollectionViewCell: UICollectionViewCell {
    public override init(frame: CGRect) {
        super.init(frame: frame)
        autoExecute()
    }
    
    public required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    open var handleBlock:((_ eventName:String, _ userInfo:[String:Any]) -> Void)? = nil
    
    open func autoExecute() {
        
    }
    
    open func didSetupUI(view: UIView) -> Void {
        
    }
    
    open func didBindData() -> Void {
        
    }
    
    open func configData(viewModel:YKViewModel, indexPath:IndexPath, dataSource:[Any]) ->Void {
        
    }
    
}
